src/core/utils/fileutils.cpp

```diff
@@ -174,6 +174,38 @@ QByteArray FileUtils::fileChecksum( const QString &fileName, const QCryptographi
   return QByteArray();
 }
 
+QString FileUtils::fileEtag( const QString &fileName, int partSize )
+{
+  QFile f( fileName );
+  if ( !f.open( QFile::ReadOnly ) )
+    return QString();
+
+  const qint64 fileSize = f.size();
+  QCryptographicHash hash( QCryptographicHash::Md5 );
+  if ( fileSize <= partSize )
+  {
+    if ( hash.addData( &f ) )
+    {
+      return hash.result().toHex();
+    }
+  }
+  else
+  {
+    QByteArray md5SumsData;
+    qint64 readSize = 0;
+    while ( readSize < fileSize )
+    {
+      hash.addData( f.read( partSize ) );
+      md5SumsData += hash.result();
+      hash.reset();
+      readSize += partSize;
+    }
+    hash.addData( md5SumsData );
+    return QStringLiteral( "%1-%2" ).arg( hash.result().toHex() ).arg( readSize / partSize );
```

Comment:
```suggestion
    return QStringLiteral( "%1-%2" ).arg( hash.result().toHex() ).arg( readSize / partSize );
```

I am only uncofortable with this division here. I hope this will never result into `<md5>-3.000004`?


We can either round, or just do a `Math.ceil(fileSize / partSize)`, or just add an int counter when looping? It's an implementation detail, but I would either assert `readSize == fileSize` or remove the `readSize` at all.

Reply:
@suricactus , the readSize will most often be greater than fileSize here. The readSize is set to 0 then partSize is added every time it reads a chunk. The last chunk will rarely be a perfect partSize, so the readSize will be greater. But it also means the division will _always_ be a perfect integer :)

Reply:
```suggestion
    return QStringLiteral( "%1-%2" ).arg( hash.result().toHex() ).arg( chunksCount );
```

I think this would be better without the need to divide, floats make me feel unconfortable.

Reply:
Discard my previous comment, it was with the wrong assumption that `int/int=float` in c++.